modules/@apostrophecms/schema/index.js

```diff
@@ -611,92 +616,207 @@ module.exports = {
-                errors.push(invalid);
-              } else {
-                error.path = field.name;
-                errors.push(error);
+          try {
+            const isRequired = await self.isFieldRequired(req, field, destination);
+            await convert(
+              req,
+              {
+                ...field,
+                required: isRequired
+              },
+              data,
+              destination,
+              {
+                ...options,
+                rootConvert: false
               }
-            }
+            );
+          } catch (err) {
+            const error = Array.isArray(err)
+              ? self.apos.error('invalid', { errors: err })
+              : err;
+
+            error.path = field.name;
+            error.schemaPath = field.aposPath;
+            convertErrors.push(error);
           }
         }
 
-        const errorsList = [];
-
-        for (const error of errors) {
-          if (error.path) {
-            // `self.isVisible` will only throw for required fields that have
-            // an external condition containing an unknown module or method:
-            const isVisible = isParentVisible === false
-              ? false
-              : await self.isVisible(req, schema, destination, error.path);
-
-            if (!isVisible) {
-              // It is not reasonable to enforce required,
-              // min, max or anything else for fields
-              // hidden via "if" as the user cannot correct it
-              // and it will not be used. If the user changes
-              // the conditional field later then they won't
-              // be able to save until the erroneous field
-              // is corrected
-              const name = error.path;
-              const field = schema.find(field => field.name === name);
-              if (field) {
-                // To protect against security issues, an invalid value
-                // for a field that is not visible should be quietly discarded.
-                // We only worry about this if the value is not valid, as otherwise
-                // it's a kindness to save the work so the user can toggle back to it
-                destination[field.name] = klona((field.def !== undefined)
-                  ? field.def
-                  : self.fieldTypes[field.type]?.def);
-                continue;
-              }
+        if (!rootConvert) {
+          if (convertErrors.length) {
+            throw convertErrors;
+          }
+
+          return;
+        }
+
+        const nonVisibleFields = await self.getNonVisibleFields({
+          req,
+          schema,
+          destination
+        });
+
+        const validErrors = await self.handleConvertErrors({
+          req,
+          schema,
+          convertErrors,
+          destination,
+          nonVisibleFields
+        });
+
+        for (const error of validErrors) {
+          self.apos.util.error(error.stack);
+        }
```

Comment:
Moved log of errores out of `handleErrors` method.
Created a ticket to improve globally message printed from convert errors.